core/arch/arm/mm/sp_mem.c

```diff
@@ -53,11 +53,34 @@ struct sp_mem *sp_mem_new(void)
 void sp_mem_remove(struct sp_mem *smem)
 {
 	uint32_t exceptions = 0;
+	int i = 0;
 
-	exceptions = cpu_spin_lock_xsave(&sp_mem_lock);
-	if (smem) {
-		SLIST_REMOVE(&mem_shares, smem, sp_mem, link);
-		free(smem);
+	if (!smem)
+		return;
+
+	/* Remove all receivers */
+	while (!SLIST_EMPTY(&smem->receivers)) {
```

Comment:
Can we guarantee that this thread has exclusive access to `smem` and all its fields here?
There isn't any risk that some other thread might be doing a `FFA_MEM_RETRIEVE_REQ` on this at the same time?